metals/src/main/scala/scala/meta/internal/metals/ScalaVersions.scala

```diff
@@ -97,4 +97,35 @@ object ScalaVersions {
       case _ => Scala213
     }
   }
+
+  private val scalaVersionRegex =
+    "(_?)(\\d)\\.(\\d{1,2})(\\.\\d(-(RC|M)\\d)?)?".r
+
+  def scalaBinaryVersionFromJarName(filename: String): Option[String] = {
+    val dropEnding = filename
+      .stripSuffix(".jar")
+
+    scalaVersionRegex
+      .findAllMatchIn(dropEnding)
+      .toList
+      .flatMap { m =>
+        val weight = if (Option(m.group(1)).isDefined) 0 else 1
+        val major = m.group(2)
+        val minor = m.group(3)
+        val ending = Option(m.group(4)).map(s => s"$s").getOrElse("")
+        val version = s"$major.$minor$ending"
```

Comment:
Added doc.

> Get the version suffix

Hypothetically there might be some artifacts which version matches on scala version.
Smth like `library-name_3.0.0-RC1-2.12` .
And it's anyway unclear where is the version suffix starts in case if artifact name has `-` symbols